app/graphql/resolvers/subscription_resolver.rb

```diff
@@ -9,12 +9,18 @@ class SubscriptionResolver < Resolvers::BaseResolver
 
     description "Query a single subscription of an organization"
 
-    argument :id, ID, required: true, description: "Uniq ID of the subscription"
+    argument :external_id, ID, required: false, description: "External ID of the subscription"
+    argument :id, ID, required: false, description: "Lago ID of the subscription"
 
     type Types::Subscriptions::Object, null: true
 
-    def resolve(id: nil)
-      current_organization.subscriptions.find(id)
+    def resolve(id: nil, external_id: nil)
+      if id.nil? && external_id.nil?
+        raise GraphQL::ExecutionError, "You must provide either `id` or `external_id`."
+      end
+
+      return current_organization.subscriptions.find(id) if id.present?
+      current_organization.subscriptions.find_by!(external_id:)
```

Comment:
As multiple subscription could match the external id, maybe we could use something similar to what we do in the `Event#subscription` method:

```ruby
current_organization.subscriptions
  .where(external_id:)
  .order("terminated_at DESC NULLS FIRST, started_at DESC")
  .first
```
It could make the result more predictable by always returning first the currently active subscription or the most recently terminated one.


What do you think?

Reply:
By the way @vincent-pochet @rsempe , I recently noticed the same behaviour on the API when updating the documentation: 

https://github.com/getlago/lago-openapi/blob/main/src/resources/subscription.yaml#L21

I guess we could update the behaviour there as well. What do you think

Reply:
It worth a look indeed